### DISCUSSION TITLE
Implement OnDisconnect (build a presence system)

### DISCUSSION THREAD

**Discussion by george-kar** [SchneiderInfosystems/FB4D#149] [Q&A]
My web project already uses Firebase RealTime OnDisconnect 

https://firebase.google.com/docs/reference/js/v8/firebase.database.OnDisconnect class.

I was wondering if thats possible with Delphi implementation. If not, if its hard to added?

If I can use that in my application then I can use the component and of course I will donate.

**Comment by SchneiderInfosystems** [comment]
Do you want to build a presence system with it?

As it is described in the following blog: https://firebase.blog/posts/2013/06/how-to-build-presence-system?

In the REST API reference I can't find anything about "OnDisconnect", so I would have to reverse engineer a JavaScript solution and see what is sent to the server with this command. 

Do you already have a simple JS solution that you can publish here or which you could send me by mail?

**Comment by george-kar** [comment]
Hello,

Yes thats what I want to create a presence system as described on this link you provided. But you are right there isn't on the REST API. But its easy to reverse it, I had a quick look and its possible.

Here is a working example in Javascript that would help you to see in action:  https://github.com/isarantis/firebase-presence-example

**Comment by SchneiderInfosystems** [comment]
Thank you for this example. 

Next week I will be on vacation in the Swiss mountains for a week. After that I will check what the effort for the implementation of this new feature is. Probably I won't be able to implement it in the first quarter of 2023 without a donation in the range between 1000 to 3000 EUR, because otherwise I have to implement more urgent projects first.

**Reply by george-kar** [reply]
Hello, I will email you asap.

**Reply by SchneiderInfosystems** [reply]
You can use this mail address: 
fb4d@schneider-infosys.ch
